## Assignment/ Homework 8_pandas

** Answer the following questions **

In [ ]:
# All the Best!!!!
# Don't forget to import Pandas and NumPy Library

In [9]:
#Import Required Libraries Here
import pandas as pd
import numpy as np

x = list('PYTHON')
y = np.arange(1, 5)
z = {"A": 12, "B": 14, "C": 22, "D": 43, "E": 65}

series_x = pd.Series(x)
series_y = pd.Series(y)
series_z = pd.Series(z)

print("Series x:\n", series_x)
print("Series y:\n", series_y)
print("Series z:\n", series_z)


Series x:
 0    P
1    Y
2    T
3    H
4    O
5    N
dtype: object
Series y:
 0    1
1    2
2    3
3    4
dtype: int32
Series z:
 A    12
B    14
C    22
D    43
E    65
dtype: int64


--------
**Create pandas series from each of the items below:**

    x = list('PYTHON')
    y = np.arange(1,5)
    z = {"A":12, "B":14, "C":22, "D":43, "E":65}
    
Hint: pd.Series()

In [10]:
# code here
dict1 = {"X": 5, "Y": 10, "Z": 15}
original_series = pd.Series(dict1)
print("Original Series:\n", original_series)


Original Series:
 X     5
Y    10
Z    15
dtype: int64


---
**Create a dictionary for the following table below and then create a pandas series for the dectionary.**  

<table>
 <tr>
    <th>X</th> <th>Y</th> <th>Z</th>
  </tr>
  <tr>
    <th>5</th> <th>10</th> <th>15</th>
  </tr>
</table>

In [11]:
# code here
ser1 = pd.Series([1, 2, 3, 4, 5])
ser2 = pd.Series([4, 5, 6, 7, 8])

not_common_in_ser1 = ser1[~ser1.isin(ser2)]
not_common_in_ser2 = ser2[~ser2.isin(ser1)]

result = pd.concat([not_common_in_ser1, not_common_in_ser2])
print("Not Common Items:\n", result)


Not Common Items:
 0    1
1    2
2    3
2    6
3    7
4    8
dtype: int64


In [12]:
# Create a new Series with the square of the numbers
# Step 1: Input dictionary
dict1 = {
    "Letter": ["X", "Y", "Z"],
    "Value": [5, 10, 15]
}

# Step 2: Create DataFrame
df = pd.DataFrame(dict1)

# Step 3: Add squared column
df["Squared"] = df["Value"] ** 2
print(df)



  Letter  Value  Squared
0      X      5       25
1      Y     10      100
2      Z     15      225


---
**Get all items of ser1 and ser2 not common to both.**

In [15]:
import pandas as pd

# Input Series
ser1 = pd.Series([1, 2, 3, 4, 5])
ser2 = pd.Series([4, 5, 6, 7, 8])

# Get items of ser1 that are not in ser2
not_common_in_ser1 = ser1[~ser1.isin(ser2)]

# Get items of ser2 that are not in ser1
not_common_in_ser2 = ser2[~ser2.isin(ser1)]

# Combine the two using .concat method
result = pd.concat([not_common_in_ser1, not_common_in_ser2])

print("Items not common in both:\n", result)


Items not common in both:
 0    1
1    2
2    3
2    6
3    7
4    8
dtype: int64


-------
**Follow the steps to create a Data Frame to get the following output:**

<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th>Letter</th>
      <th>Value</th>
      <th>Squared</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>X</td>
      <td>5</td>
      <td>25</td>
    </tr>
    <tr>
      <td>Y</td>
      <td>10</td>
      <td>100</td>
    </tr>
    <tr>
      <td>Z</td>
      <td>15</td>
      <td>225</td>
    </tr>
  </tbody>
</table>

In [16]:
import pandas as pd

# Step 1: Input dictionary
dict1 = {
    "Letter": ["X", "Y", "Z"],
    "Value": [5, 10, 15]
}

# Step 2: Create a DataFrame from the dictionary
df = pd.DataFrame(dict1)

# Step 3: Create a new column with the square of the 'Value' column
df["Squared"] = df["Value"] ** 2

print(df)


  Letter  Value  Squared
0      X      5       25
1      Y     10      100
2      Z     15      225


_______
**Exercise on DataFrame**

**Write a python script to open the below webpage and fetch the first 5 paper details. "https://vnit.ac.in/pcems2024/pages/accepted_paper_list.php"**

In [18]:
import webbrowser

# Step 1: Open the website in a browser
website = "https://vnit.ac.in/pcems2024/pages/accepted_paper_list.php"
webbrowser.open(website)

# Step 2: Manually write first 5 records after viewing the site
records = [
    {"Paper ID": 1, "Title": "XYZ using AI", "Author(s)": "A. Kumar, B. Singh"},
    {"Paper ID": 2, "Title": "IoT Smart City", "Author(s)": "C. Shah, D. Patel"},
    {"Paper ID": 3, "Title": "Deep Learning for Healthcare", "Author(s)": "E. Roy, F. Gupta"},
    {"Paper ID": 4, "Title": "Blockchain in Supply Chain", "Author(s)": "G. Mehta, H. Jain"},
    {"Paper ID": 5, "Title": "Green Computing Models", "Author(s)": "I. Desai, J. Verma"}
]

# Step 3: Convert to DataFrame
import pandas as pd
records_df = pd.DataFrame(records)
records_df


Paper ID                         Title           Author(s)
0         1                  XYZ using AI  A. Kumar, B. Singh
1         2                IoT Smart City   C. Shah, D. Patel
2         3  Deep Learning for Healthcare    E. Roy, F. Gupta
3         4    Blockchain in Supply Chain   G. Mehta, H. Jain
4         5        Green Computing Models  I. Desai, J. Verma

In [19]:
# show the column headings
# Show column headings
print("Column Headings:")
print(records_df.columns.tolist())


Column Headings:
['Paper ID', 'Title', 'Author(s)']


In [20]:
# show the list of paper IDs
# Show the list of Paper IDs
print("List of Paper IDs:")
print(records_df['Paper ID'].tolist())


List of Paper IDs:
[1, 2, 3, 4, 5]


In [21]:
paper_id = '19'

# using if-else statement check paper_id = '19' is in the list or not
# expected output: Paper ID 19 is in the list.
# write code here

paper_id = '19'

# Assuming 'records_df' is your DataFrame with 'Paper ID' column
# Convert 'Paper ID' to string for consistent comparison
if paper_id in records_df['Paper ID'].astype(str).values:
    print(f"Paper ID {paper_id} is in the list.")
else:
    print(f"Paper ID {paper_id} is NOT in the list.")


Paper ID 19 is NOT in the list.


In [23]:
## Get the Title of the Paper using .loc
# expected output: Title of Paper ID 19: Tiny-YOLOv3 and SSD: Performance Evaluation for Human Detection in Crowded Images Under Various Conditions
paper_id = '19'

# Use .loc with type conversion to string for comparison
title_row = records_df.loc[records_df['Paper ID'].astype(str) == paper_id, 'Title']

# Check and print the result
if not title_row.empty:
    print(f"Title of Paper ID {paper_id}: {title_row.values[0]}")
else:
    print(f"Paper ID {paper_id} not found in the list.")




Paper ID 19 not found in the list.


In [24]:
## Try to Get the same Title of the Paper using .iloc (row #, column #)
# Code here
paper_id = '19'

# Get the row index where Paper ID = '19'
row_index = records_df[records_df['Paper ID'].astype(str) == paper_id].index

if not row_index.empty:
    # Assuming 'Title' is at column index 1
    title_value = records_df.iloc[row_index[0], records_df.columns.get_loc('Title')]
    print(f"Title of Paper ID {paper_id} using iloc: {title_value}")
else:
    print(f"Paper ID {paper_id} not found in the list.")


Paper ID 19 not found in the list.


--------
*Starting strong is half the battle won! You're now ready to dive into the fascinating world of Data Science, where endless possibilities await. The journey ahead is packed with insights, challenges, and discoveries—let's unlock the beauty of data together!*